ui/tests/acceptance/open-api-path-help-test.js

```diff
@@ -0,0 +1,593 @@
+import { module, test } from 'qunit';
+import Model from '@ember-data/model';
+import { setupApplicationTest } from 'vault/tests/helpers';
+import authPage from 'vault/tests/pages/auth';
+import { deleteAuthCmd, deleteEngineCmd, mountAuthCmd, mountEngineCmd, runCmd } from '../helpers/commands';
+import openApiDrivenAttributes from '../helpers/openapi-driven-attributes';
+
+/**
+ * This set of tests is for ensuring that backend changes to the OpenAPI spec
+ * are known by UI developers and adequately addressed in the UI. When changes
+ * are detected from this set of tests, they should be updated to pass and
+ * smoke tested to ensure changes to not break the GUI workflow.
+ */
+module('Acceptance | OpenAPI path help test', function (hooks) {
+  setupApplicationTest(hooks);
+  hooks.beforeEach(function () {
+    this.pathHelp = this.owner.lookup('service:pathHelp');
+    this.newModel = Model.extend({});
+    return authPage.login();
+  });
+
+  module('engine: ssh role', function (hooks) {
+    hooks.beforeEach(async function () {
+      this.backend = 'ssh-openapi';
+      await runCmd(mountEngineCmd('ssh', this.backend), false);
+    });
+    hooks.afterEach(async function () {
+      await runCmd(deleteEngineCmd(this.backend), false);
+    });
+
+    test('getProps returns correct model attributes', async function (assert) {
+      const helpUrl = `/v1/${this.backend}/roles/example?help=1`;
+      const result = await this.pathHelp.getProps(helpUrl, this.backend);
+      assert.deepEqual(result, openApiDrivenAttributes.sshRole);
+    });
+  });
+
+  module('engine: kmip enterprise', function (hooks) {
+    hooks.beforeEach(async function () {
+      this.backend = 'kmip-openapi';
+      await runCmd(mountEngineCmd('kmip', this.backend), false);
+    });
+    hooks.afterEach(async function () {
+      await runCmd(deleteEngineCmd(this.backend), false);
+    });
+
+    test('getProps returns correct model attributes', async function (assert) {
+      const config = await this.pathHelp.getProps(`/v1/${this.backend}/config?help=1`, this.backend);
+      assert.deepEqual(config, openApiDrivenAttributes.kmipConfig, 'kmip config');
+
+      const role = await this.pathHelp.getProps(
+        `/v1/${this.backend}/scope/example/role/example?help=1`,
+        this.backend
+      );
+      assert.deepEqual(role, openApiDrivenAttributes.kmipRole, 'kmip role');
+    });
+  });
```

Comment:
I wonder if this could be defined as a test [helper](https://github.com/hashicorp/vault/blob/main/ui/tests/unit/adapters/oidc/test-helper.js)

(like Jordan did for oidc [adapters](https://github.com/hashicorp/vault/blob/main/ui/tests/unit/adapters/oidc/assignment-test.js#L26)) which would expedite adding test coverage in the future if/when we add models that leverage openApi

Reply:
Took this suggestion and made 2 test helpers, one for secrets engines and one for auth methods. I think it also addressed some of the concerns you had about naming since the object keys are now more conventional based on the backend / modelName. Let me know what you think!

Reply:
Yeah - that looks great!